app/src/main/kotlin/com/saltedge/authenticator/features/connections/list/ConnectionsListCollector.kt

```diff
@@ -54,6 +54,19 @@ fun List<Connection>.convertConnectionsToViewModels(context: Context, locationMa
     return this.map { connection -> connection.convertConnectionToViewModel(context, locationManager) }
 }
 
+/**
+ * Check if location permissions are granted and geolocation is required
+ *
+ * @return Boolean, true Connection geolocation is required and locations permissions are granted
+ */
+fun checkGrantAccessToLocationData(geolocationRequired: Boolean?, locationPermissionsAreGranted: Boolean): Boolean {
```

Comment:
I understand, that specs will be in the next pull request. But, this function has many logic operations and this function is the base of the geolocation task. According to the description of this function is not clear what exactly this function should return. Also, spec will help you to write correct description.